src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -338,16 +340,14 @@ internal VisualStudioProjectTracker ProjectTracker
         // Solution Explorer in the SolutionExplorerShim, where if we just could more directly get to the rule set file it'd simplify this.
         internal override string? TryGetRuleSetPathForProject(ProjectId projectId)
         {
-            using (_gate.DisposableWait())
+            // _projectToRuleSetFilePath is immutable, so can be used outside of locks
```

Comment:
this always worries me as i always think there's .net subtleties about all writes being observed.  though i think you'r correct about safety.